torch/nn/modules/module.py

```diff
@@ -193,13 +193,44 @@ def _apply(self, fn):
         for module in self.children():
             module._apply(fn)
 
-        for param in self._parameters.values():
+        def compute_should_use_set_data(tensor, tensor_applied):
+            if torch._has_same_tensorimpl_type(tensor, tensor_applied):
+                # If the new tensor has the same TensorImpl type as the existing tensor,
+                # the current (incorrect) behavior is to change the tensor in-place using `.data =`,
```

Comment:
This comment just begs the question: why is it incorrect?  If it's incorrect, why don't we change it immediately?